File: server/src/service/auth_service.py
from database import SessionLocal
from model import User
from util.helper import Helper

db = SessionLocal()


class AuthService:
    @staticmethod
    def login(email: str, password: str):
        user = db.query(User).filter_by(email=email).first()
        if user is None or not Helper.check_password(password, user.password):
            return None
        access_token = Helper.generate_access_token({"user_id": user.id})
        refresh_token = Helper.generate_refresh_token({"user_id": user.id})
        return {"access_token": access_token, "refresh_token": refresh_token}

    @staticmethod
    def register(payload: dict):
        user = User(
            name=payload.name,
            email=payload.email,
            password=Helper.generate_hash_password(payload.password),
        )
        db.add(user)
        db.commit()
        return user

File: server/service/auth-service.js
import client from "../db.js"
import Helper from "../util/helper.js"

const helper = new Helper();
class AuthService {


    async login(email, pasword) {
        const text =
            `
        SELECT * FROM users WHERE email = $1 and password = $2

        `
        try {
            const user = client.query(text, [email, pasword]);
            if (user.rows[0]) {
                const payload = {
                    id: user.id,
                    email: user.email
                }
                return {
                    accessToken: helper.generateAccessToken({ payload }),
                    refreshToken: helper.generateRefreshToken({ payload })
                }
            } else {
                throw new Error("Account not found!")
            }
        } catch (e) {
            return e;
        }
    }

    async register() {

    }
}


export default AuthService